snuba/datasets/profiles_processor.py

```diff
@@ -9,48 +9,92 @@
 
 metrics = MetricsWrapper(environment.metrics, "profiles.processor")
 
-RETENTION_DAYS_ALLOWED = frozenset([30, 90])
-
 
 class ProfilesMessageProcessor(MessageProcessor):
     def process_message(
         self, message: Mapping[str, Any], metadata: KafkaMessageMetadata
     ) -> Optional[ProcessedMessage]:
         try:
-            retention_days = message["retention_days"]
-            if retention_days not in RETENTION_DAYS_ALLOWED:
-                retention_days = 30
```

Comment:
Please do not remove this validation. If you allow users to drive the retention, the work of TTL merges on Clickhouse would increase since each partition could have parts which might require TTL merge. That increases load on you cluster unnecessarily. I would advise against doing so.

Reply:
Sure, I'll add it back. It's still not users driving the retention but Sentry via the quota configured by their plan. This logic is just too simple.

I'll use whatever you use in transactions instead.